### PR TITLE
feat(qualtrics methods): displayQualtricsIntercept, setQualtricsProperties, isQualtricsInterceptAvailableForUser

### LINKED ISSUE #178: New bridge method to show Qualtrics survey
**Message type**: `DISPLAY_QUALTRICS_INTERCEPT`

**Originator**: Web

**Min app version**: 24.12

## Description

In order to accomplish the integration of the Qualtrics SDK for VIVO ([APPS-8203](https://jira.tid.es/browse/APPS-8203)) we need to implement a new bridge method to show (if possible) a suervey to the user using this SDK.

This new method will provide an interceptor id in the request. If provided the app will try to show the survey related to this intercept id.

It will return a boolean indicating if the survey has been displayed or not (

## Differences between iOS/Android (if any)

N/A

## Payload definition

**Request Payload**
```ts
{
    intercetptId: string;
}
```

Examples:

```json
{
    "type": "DISPLAY_QUALTRICS_INTERCEPT",
    "id": "id-display-qualtrics-intercept",
    "payload": {
        "intercetptId": "SI_81gUpIN3UhXAZU2"
    }
}
```

**Response Payload**
```ts
{
    displayed: boolean;
}
```

Examples:

```json
{
    "type": "DISPLAY_QUALTRICS_INTERCEPT",
    "id": "id-display-qualtrics-intercept",
    "payload": {
        "displayed": true
    }
}
```

## Possible error cases

500 → Internal Error: If an error occurred invoking the SDK
501 → SDK not initialized

### LINKED ISSUE #179: New bridge method to set Qualtrics properties
**Message type**: `SET_QUALTRICS_PROPERTIES`

**Originator**: Web

**Min app version**: 24.12

## Description

In order to accomplish the integration of the Qualtrics SDK for VIVO ([APPS-8203](https://jira.tid.es/browse/APPS-8203)) we need to implement a new bridge method to set properties in Qualtrics SDK.

Due to the SDK has three different methods to set properties depending on its type and to avoid having multiple bridge methods, tha approach is passing in the request a struct containing three arrays for each property type: string, number and dateTime. This way, this method will allow setting multiple properties with one call and avoids using generic types and casting that may produce undesired errors.

Number and string properties require a value of the corresponding type (number or string) and a key to identify the property where the value is going to be set.

Date time properties only require a key to identify the property where the value is going to be set. This value will be a timestamp.

## Differences between iOS/Android (if any)

N/A

## Payload definition

**Request Payload**  

```ts
{
    stringProperties: [{key: string, value: string}],
    numberProperties:  [{key: string, value: number}],
    dateTimePropertyKeys: string[]
}
```

Examples:

```json
{
    "type": "SET_QUALTRICS_PROPERTIES", 
    "id": "id-set-qualtrics-properties",
    "payload": {
        "stringProperties": [
            {
                "key": "stringProperty1",
                "value": "Test value 1"
            },
            {
                "key": "stringProperty2",
                "value": "Test value 2"
            },
        ],
        "numberProperties": [
            {
                "key": "numericProperty",
                "value": 12.0
            }
        ],
        "dateTimePropertyKeys": ["dateTimePropertyKey"]
    }
}
```

**Response Payload**  
No payload required

Examples:

```json
{
    "type": "SET_QUALTRICS_PROPERTIES", 
    "id": "id-set-qualtrics-properties"
}
```

## Possible error cases

500 → Internal Error: If an error occurred invoking the SDK
501 → SDK not initialized

### LINKED ISSUE #184: New bridge method to check if an intercept id is available for user
**Message type**: `IS_QUALTRICS_INTERCEPT_AVAILABLE_FOR_USER`

**Originator**: web

**Min app version**: 24.12

## Description

In order to accomplish the integration of the Qualtrics SDK for VIVO ([APPS-8203](https://jira.tid.es/browse/APPS-8203)) we need to implement a new bridge method to check if a given intercept is available for the user.

## Differences between iOS/Android (if any)

N/A

## Payload definition

**Request Payload**  

```ts
{
    interceptId: string;
}
```

Examples:

```json
{
    "type": "IS_QUALTRICS_INTERCEPT_AVAILABLE_FOR_USER", 
    "id": "id-is-qualtrics-intercept-available-for-user",
    "payload": {
        "intercetptId": "SI_81gUpIN3UhXAZU2"
    }
}
```

**Response Payload**
```ts
{
    isAvailable: boolean;
}
```

Examples:

```json
{
    "type": "IS_QUALTRICS_INTERCEPT_AVAILABLE_FOR_USER",
    "id": "id-is-qualtrics-intercept-available-for-user",
    "payload": {
        "isAvailable": true
    }
}
```

## Possible error cases

500 → SDK not initialized
500 → Internal Error: If an error occurred invoking the SDK

### PR DESCRIPTION
**Pull request by atabel**
https://jira.tid.es/browse/WEB-2119

closes #178 
closes #179 
closes #184

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1479,6 +1479,93 @@ validateDatamobRequirements: ({phoneNumber: string, tokenPassword: string}) => P
 
 -   `requirements`: A map with the requirements.
 
+### displayQualtricsIntercept
+
+<kbd>App version >=24.12</kbd>
```
Comment (dpastor) [inline README.md:1484]: All methods will be available only for iOS by the moment. In android it's expected to be integrated at some next version.
We can just a "Only for iOS" comment, and when it is integrated in android we can update the doc with the proper version.

**Review by pladaria — APPROVED** [review]

File: src/qualtrics.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+export const setQualtricsProperties = ({
+    stringProperties,
+    numberProperties,
+    dateTimePropertyKeys,
+}: {
+    stringProperties: Array<QualtricsProperty<string>>;
+    numberProperties: Array<QualtricsProperty<number>>;
+    dateTimePropertyKeys: Array<string>;
```
Comment (dpastor) [inline src/qualtrics.ts:25]: I see we didn't specified that in the spec, but probably is interesting to give these properties a default value (empty array) from js side?

Comment (atabel) [reply src/qualtrics.ts:25]: lgtm, done

**Review by dpastor — APPROVED** [review]
Thanks a lot! Please just check this comment -> https://github.com/Telefonica/webview-bridge/pull/185#discussion_r1881609315

**Pull request merged**